witness/golang/cmd/witness/internal/witness/witness.go

```diff
@@ -33,37 +35,29 @@ type Chkpt struct {
 	Raw  []byte
 }
 
-// ChkptStorage is an interface for reading and writing checkpoints.
-type ChkptStorage interface {
-	// GetLatest returns the latest checkpoint for a given log.
-	GetLatest(logID string) (Chkpt, error)
-
-	// SetCheckpoint adds a checkpoint to the storage for a given log if the
-	// provided input latestSize is the size of its largest stored checkpoint.
-	SetCheckpoint(ctx context.Context, logID string, latestSize uint64, c Chkpt) error
+// Database reads from and writes checkpoints to persistent storage.
+type Database struct {
```

Comment:
I wonder whether you need this type. Its "private"/unexported methods could be made methods of the `Witness`, or just functions accepting `db *sql.DB` as a first parameter.

Reply:
Agreed, it's now removed.